docs/frameworks/sveltekit.md

```diff
@@ -0,0 +1,229 @@
+with the same codebase.
+
+As an example using [@sveltejs/adapter-static](https://github.com/sveltejs/kit/tree/master/packages/adapter-static) <outbound-link />
+with `generateSW` strategy and `Prompt for update` behavior, you will need:
+
+<details>
+<summary>1) add <strong>pwa.js</strong> script</summary>
+
+```js
+import { resolveConfig } from 'vite'
+import { VitePWA } from 'vite-plugin-pwa';
+import { pwaConfiguration } from './pwa-configuration.js';
+import { copyFileSync } from 'fs';
+
+const webmanifestDestinations = [
+	'./.svelte-kit/output/client/',
+	'./build/',
+]
+
+const swDestinations = [
+	'./build/',
+]
+
+const buildPwa = async() => {
+  // const pwaConfiguration = await buildPwaConfiguration()
+  const config = await resolveConfig({ plugins: [VitePWA({ ...pwaConfiguration })] }, 'build', 'production' )
+  // when `vite-plugin-pwa` is presented, use it to regenerate SW after rendering
+  const pwaPlugin = config.plugins.find(i => i.name === 'vite-plugin-pwa')?.api
+  if (pwaPlugin?.generateSW) {
+    console.log('Generating PWA...')
+    await pwaPlugin.generateSW()
+    webmanifestDestinations.forEach(d => {
+      copyFileSync('./.svelte-kit/output/client/_app/manifest.webmanifest', `${d}/manifest.webmanifest`)
+    })
+    // don't copy workbox, svelte kit will copy it
+    swDestinations.forEach(d => {
+      copyFileSync('./.svelte-kit/output/client/sw.js', `${d}/sw.js`)
+    })
+    console.log('Generated PWA complete')
+  }
+} 
+
+buildPwa()
+```
+</details>
+
+
+<details>
+<summary>2) add <strong>pwa-configuration.js</strong> script</summary>
+
+```js
+const pwaConfiguration = {
+  srcDir: './build',
+  outDir: './.svelte-kit/output/client',
+  includeManifestIcons: false,
+  base: '/',
+  scope: '/',
+  manifest: {
+    short_name: "<YOUR APP SHORT NAME>",
+    name: "<YOUR APP NAME>",
+    scope: "/",
+    start_url: "/",
+    display: "standalone",
+    theme_color: "#ffffff",
+    background_color: "#ffffff",
+    icons: [
+      {
+        src: "/pwa-192x192.png",
+        sizes: "192x192",
+        type: "image/png"
+      },
+      {
+        src: "/pwa-512x512.png",
+        "sizes": "512x512",
+        "type": "image/png"
+      },
+      {
+        src: "/pwa-512x512.png",
+        "sizes": "512x512",
+        "type": "image/png",
+        purpose: 'any maskable'
+      }
+    ]
+  },
+  workbox: {
+    // mode: 'development',
+    navigateFallback: '/',
+    // vite and sveltekit are not aligned: pwa plugin will use /\.[a-f0-9]{8}\./ by default: #164 optimize workbox work
+    dontCacheBustURLsMatching: /-[a-f0-9]{8}\./,
+    globDirectory: './build/',
+    globPatterns: ['robots.txt', '**/*.{js,css,html,ico,png,svg,webmanifest}'],
+    globIgnores: [
+      '**/sw*', '**/workbox-*'
+    ],
+    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+    manifestTransforms: [async(entries) => {
+      // manifest.webmanifest is added always by pwa plugin, so we remove it
```

Comment:
is this something that should be requested to be changed in the PWA plugin so that it doesn't have to be handled here? maybe what can't be put in the PWA plugin could be provided as a `defaultSvelteKitTransform` or something so that there's not so much that needs to be copy and pasted into each project?

Reply:
This is an example using `static adapter`, we just traverse generated html files and chnage the name since SvelteKit use that name for the route name, we just need to modify the url, the hash is generated by workbox: this will allow to work offline.

In this callback, workbox will provide the html files with the hash calculated: change the html file name with the router name.

Reply:
that's helpful details. maybe include it as a code comment?

Reply:
Do you mean to include my response on the code? Tmr i will include it

Reply:
Yes, I thought it might be useful to include something similar not your response